### DISCUSSION TITLE
Running UFS test case on a level 3/4 system - rt.sh framework

### DISCUSSION THREAD

**Discussion by jaberg7777** [ufs-community/ufs-weather-model#2237] [Q&A] [answered]
Hi all - I'm an undergraduate researcher at Northwestern University, and I've been attempting to set up UFS on our HPC cluster. I originally used the simple test case framework posted on the UFS wiki (which worked), but following the updates that took place several weeks ago I have struggled to replicate the test case (linked below): 

https://github.com/ufs-community/ufs-weather-model/wiki/Getting-Started

I've made a series of modifications to the referenced scripts and the model code now compiles. However, I run into an issue in the run stage with these three lines of code in the rt.sh script: 

NPUTDATA_ROOT=${INPUTDATA_ROOT:-$DISKNM/NEMSfv3gfs/input-data-20221101}
INPUTDATA_ROOT_WW3=${INPUTDATA_ROOT}/WW3_input_data_20220624
INPUTDATA_ROOT_BMIC=${INPUTDATA_ROOT_BMIC:-$DISKNM/NEMSfv3gfs/BM_IC-20220207}

They seem to reference directories that (I'm guessing) are internally available on NOAA systems, but are not publicly documented.

I was wondering if these are directories that one can get access to or if there is an alternate way to go about this entirely. Thanks a ton!

**Comment by NickSzapiro-NOAA** [comment, marked as answer]
Hi. Input files must be provided for level 3-4 systems. Documentation here may help:
https://ufs-weather-model.readthedocs.io/en/develop/BuildingAndRunning.html#get-data

You can access current INPUTDATA_ROOT publicly at https://noaa-ufs-regtests-pds.s3.amazonaws.com/index.html#input-data-20221101/

**Reply by gspetro-NOAA** [reply]
Yes, @jaberg7777 input data is publicly available in the WM RT data bucket at the link @NickSzapiro-NOAA provided. The BM_IC20220207  data is also available in the bucket at https://noaa-ufs-regtests-pds.s3.amazonaws.com/index.html#BM_IC-20220207/.
In general, a bash statement like `INPUTDATA_ROOT=${INPUTDATA_ROOT:-$DISKNM/NEMSfv3gfs/input-data-20221101}` is saying "Set INPUTDATA_ROOT to the current value of INPUTDATA_ROOT or, if unset/null, to $DISKNM/NEMSfv3gfs/input-data-20221101}." So you could just change the statement to `INPUTDATA_ROOT="/path/to/my/input/data"` (where `/path/to/my/input/data` is that actual absolute path to your input data).

**Reply by jaberg7777** [reply]
Thank you, both! That clarifies a lot.
